src/App.js

```diff
@@ -53,27 +53,32 @@ const App = ({ vms, config }) => {
 
   return (
     <Router basename={AppConfiguration.applicationURL}>
-      <div>
+      <React.Fragment>
         <VmsPageHeader page={vms.get('page')} title={fixedStrings.BRAND_NAME + ' ' + msg.vmPortal()} />
         <VerticalMenu menuItems={menu} /> { /* Disabled, to enable search for left sidebar menu */ }
+        { appReady && renderRoutes(routes) }
         <LoadingData />
-        {renderRoutes(routes)}
         <AboutDialog />
         <OptionsDialog userId={config.getIn(['user', 'id'])} />
         <OvirtApiCheckFailed />
         <TokenExpired />
-      </div>
+      </React.Fragment>
     </Router>
   )
 }
 App.propTypes = {
   vms: PropTypes.object.isRequired,
   config: PropTypes.object.isRequired,
+  appReady: PropTypes.bool,
+}
+App.defaultProps = {
+  appReady: false,
 }
 
 export default connect(
   (state) => ({
     vms: state.vms,
     config: state.config,
+    appReady: state.config.get('isFilterChecked'), // When is the app ready to display data components?
```

Comment:
Ugly imperative approach. But probably required by router.

Reply:
What would be a better way to change App render behavior based on getting the state to a particular minimum configuration?  If something like that isn't here, the first REST call can fail because the permissions and filters aren't setup yet (by the login saga)

Reply:
agree, I don't know